<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Heatmap Visualization â€¢ singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Heatmap Visualization">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
</li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cnsl_heatmap_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Heatmap Visualization</h1>
                        <h4 class="author">Yichen Wang</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/cnsl_heatmap.Rmd"><code>vignettes/cnsl_heatmap.Rmd</code></a></small>
      <div class="hidden name"><code>cnsl_heatmap.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>SingleCellTK (SCTK) is equipped with a generic heatmap plotting function called <code><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap()</a></code>, which takes an <a href="https://rdrr.io/bioc/SingleCellExperiment/man/SingleCellExperiment.html"><code>SingleCellExperiment</code></a> (SCE) object as the primary input. This generic function is a wrapper of <a href="https://rdrr.io/bioc/ComplexHeatmap/man/Heatmap.html"><code>ComplexHeatmap::Heatmap</code></a> (Zuguang Gu and <em>et. al.</em>, 2016). Similar to many heatmap plotting functions in other packages, we allow flexible legend strategies. Meanwhile, we support the direct use of information already saved in <code>colData</code> and <code>rowData</code>.</p>
</div>
<div id="examples" class="section level2">
<h2 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h2>
<p>To present the usage of <code><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap()</a></code>, we would like to use a small example provided with SCTK.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://compbiomed.github.io/sctk_docs/">singleCellTK</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"scExample"</span><span class="op">)</span> <span class="co"># This imports SCE object "sce"</span>
<span class="va">sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 200 390 
## metadata(0):
## assays(1): counts
## rownames(200): ENSG00000251562 ENSG00000205542 ... ENSG00000204472
##   ENSG00000133872
## rowData names(2): feature_ID feature_name
## colnames(390): pbmc_4k_GACCAATTCCTAGAAC-1 pbmc_4k_AAATGCCGTTTCGCTC-1
##   ... pbmc_4k_CTGAAGTTCCACTCCA-1 pbmc_4k_CATCGAACAGCCAGAA-1
## colData names(4): cell_barcode column_name sample type
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>The minimum setting for <code><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap()</a></code> is the input SCE object and the data matrix to plot (default <code>"logcounts"</code> though). In this way, all cells and features will be presented while no annotation or legend (except the main color scheme) will be shown.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap</a></span><span class="op">(</span><span class="va">sce</span>, useAssay <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></code></pre></div>
<p><img src="cnsl_heatmap_files/figure-html/hmFull-1.png" width="700"></p>
<div id="subsetting" class="section level3">
<h3 class="hasAnchor">
<a href="#subsetting" class="anchor"></a>Subsetting</h3>
<p>The first way to subset the heatmap is to directly use an index vector that can subset the input SCE object to <code>featureIndex</code> and <code>cellIndex</code>, including <code>numeric</code>, and <code>logical</code> vectors, which are widely used, and <code>character</code> vector containing the row/col names. Of course, user can directly use a subsetted SCE object as input.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Make up random downsampling numeric vector</span>
<span class="va">featureSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span>
<span class="va">cellSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap</a></span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">sce</span>, useAssay <span class="op">=</span> <span class="st">"counts"</span>, 
               featureIndex <span class="op">=</span> <span class="va">featureSubset</span>, cellIndex <span class="op">=</span> <span class="va">cellSubset</span><span class="op">)</span></code></pre></div>
<p><img src="cnsl_heatmap_files/figure-html/idxSubset-1.png" width="700"></p>
<p>In a more complexed situation, where users might only have a set of identifiers which are not inside the row/col names (i.e.Â unable to directly subset the SCE object), we provide another approach. The subset, in this situation, can be accessed via specifying a vector that contains the identifiers users have, to <code>featureIndexBy</code> or <code>cellIndexBy</code>. This specification allows directly giving one column name of <code>rowData</code> or <code>colData</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">subsetFeatureName</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">feature_name</span>, <span class="fl">50</span><span class="op">)</span>
<span class="va">subsetCellBarcode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">cell_barcode</span>, <span class="fl">50</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap</a></span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">sce</span>, useAssay <span class="op">=</span> <span class="st">"counts"</span>, 
               featureIndex <span class="op">=</span> <span class="va">subsetFeatureName</span>, featureIndexBy <span class="op">=</span> <span class="st">"feature_name"</span>,
               cellIndex <span class="op">=</span> <span class="va">subsetCellBarcode</span>, cellIndexBy <span class="op">=</span> <span class="st">"cell_barcode"</span><span class="op">)</span></code></pre></div>
<p><img src="cnsl_heatmap_files/figure-html/indexBy-1.png" width="700"></p>
</div>
<div id="annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#annotation" class="anchor"></a>Annotation</h3>
<p>As introduced before, we allow directly using column names of <code>rowData</code> or <code>colData</code> to attach color bar annotations. To make use of this functionality, pass a <code>character</code> vector to <code>rowDataName</code> or <code>colDataName</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Make up arbitrary annotation, </span>
<span class="va">rowRandLabel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">'aa'</span>, <span class="fl">100</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">'bb'</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span>
<span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">randLabel</span> <span class="op">&lt;-</span> <span class="va">rowRandLabel</span>
<span class="va">colRandLabel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">'cc'</span>, <span class="fl">195</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">'dd'</span>, <span class="fl">195</span><span class="op">)</span><span class="op">)</span>
<span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">randLabel</span> <span class="op">&lt;-</span> <span class="va">colRandLabel</span>
<span class="fu"><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap</a></span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">sce</span>, useAssay <span class="op">=</span> <span class="st">"counts"</span>, 
               featureIndex <span class="op">=</span> <span class="va">featureSubset</span>, cellIndex <span class="op">=</span> <span class="va">cellSubset</span>,
               rowDataName <span class="op">=</span> <span class="st">"randLabel"</span>, colDataName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"type"</span>, <span class="st">"randLabel"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="cnsl_heatmap_files/figure-html/colRowAnn-1.png" width="700"></p>
<p>Fully customized annotation is also supported, though it can be complexed for users. For the labeling, it is more recommanded to insert the information into <code>rowData</code> or <code>colData</code> and then make use. For coloring, information should be passed to <code>featureAnnotationColor</code> or <code>cellAnnotationColor</code>. The argument must be a <code>list</code> object with names matching the annotation classes (such as <code>"randLabel"</code> and <code>"type"</code>); each inner object under a name must be a named vector, with colors as the values and existing categories as the names. The working instance looks like this:</p>
<pre><code>## [1] "For row annotation"</code></pre>
<pre><code>## $randLabel
##        aa        bb 
## "#FF4D4D" "#4DFFFF"</code></pre>
<pre><code>## [1] "For column annotation"</code></pre>
<pre><code>## $sample
##   pbmc_4k 
## "#FF4D4D" 
## 
## $type
##      Singlet      Doublet EmptyDroplet 
##    "#4DFFFF"    "#FFC04D"    "#4D4DFF" 
## 
## $randLabel
##        cc        dd 
## "#FFFF4D" "#BA4DFF"</code></pre>
</div>
<div id="others" class="section level3">
<h3 class="hasAnchor">
<a href="#others" class="anchor"></a>Others</h3>
<p>In some cases, it might be better to do a semi-heatmap (i.e.Â split the rows/columns first and cluster them within each group) to visualize some expression pattern, such as evaluating the differential expression. For this need, use <code>rowSplitBy</code> or <code>colSplitBy</code>. Argument must be a <code>character</code> vector included inside what are annotated.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap</a></span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">sce</span>, useAssay <span class="op">=</span> <span class="st">"counts"</span>, 
               featureIndex <span class="op">=</span> <span class="va">featureSubset</span>, cellIndex <span class="op">=</span> <span class="va">cellSubset</span>,
               rowDataName <span class="op">=</span> <span class="st">"randLabel"</span>, colDataName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"type"</span>, <span class="st">"randLabel"</span><span class="op">)</span>,
               rowSplitBy <span class="op">=</span> <span class="st">"randLabel"</span>, colSplitBy <span class="op">=</span> <span class="st">"type"</span><span class="op">)</span></code></pre></div>
<p><img src="cnsl_heatmap_files/figure-html/split-1.png" width="700"></p>
<p>Text labels of features or cells can be added via <code>rowLabel</code> or <code>colLabel</code>. Use <code>TRUE</code> or <code>FALSE</code> to specify whether to show the <code>rownames</code> or <code>colnames</code> of the subsetted SCE object. Additionally, giving a single string of a column name of <code>rowData</code> or <code>colData</code> can enable the labeling of the annotation. Furthermore, users can directly throw a character vector to the parameter, with the same length of either the full SCE object or the subsetted.</p>
<p>The dendrograms for features or cells can be removed by passing <code>FALSE</code> to <code>rowDend</code> or <code>colDend</code>.</p>
<p>The row title (<code>"Genes"</code>) and column title (<code>"Cells"</code>) can be changed or removed by passing a string or <code>NULL</code> to <code>rowTitle</code> or <code>colTitle</code>, respectively.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap</a></span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">sce</span>, useAssay <span class="op">=</span> <span class="st">"counts"</span>, 
               featureIndex <span class="op">=</span> <span class="va">featureSubset</span>, cellIndex <span class="op">=</span> <span class="va">cellSubset</span>,
               rowLabel <span class="op">=</span> <span class="st">"feature_name"</span>, colLabel <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span>,
               colDend <span class="op">=</span> <span class="cn">FALSE</span>,
               rowTitle <span class="op">=</span> <span class="st">"Downsampled features"</span><span class="op">)</span></code></pre></div>
<p><img src="cnsl_heatmap_files/figure-html/label-1.png" width="700"></p>
<p>There are still some parameters not mentioned here, but they are not frequently used. Please refer to <code><a href="../reference/plotSCEHeatmap.html">?plotSCEHeatmap</a></code> as well as <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">?ComplexHeatmap::Heatmap</a></code>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
