<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Dimensionality Reduction • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Dimensionality Reduction">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
</li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ui_dimensionality_reduction_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Dimensionality Reduction</h1>
                        <h4 class="author">Irzam Sarfraz</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/ui_dimensionality_reduction.Rmd"><code>vignettes/ui_dimensionality_reduction.Rmd</code></a></small>
      <div class="hidden name"><code>ui_dimensionality_reduction.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Dimensionality Reduction tab offers a convenient way to run and visualize dimensionality reduction algorithms i.e., PCA &amp; ICA, that have been implemented from various packages and integrated together in a single interface. Additionally, high-dimensional data visualization methods such as tSNE &amp; UMAP have also been provided under the umbrella of dimensionality reduction in a separate sub-tab. The dimensionality reduction tab in divided into two separate sub-tabs, one for PCA/ICA and the other one for tSNE/UMAP, both of which offer customization options and separate visualizations.</p>
<hr>
<p>A brief summary of the two tabs is described below:</p>
<div id="pcaica" class="section level3">
<h3 class="hasAnchor">
<a href="#pcaica" class="anchor"></a>PCA/ICA</h3>
<div id="implemented-algorithms-from-packages" class="section level4">
<h4 class="hasAnchor">
<a href="#implemented-algorithms-from-packages" class="anchor"></a>Implemented Algorithms from Packages:</h4>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>reducedDims Method</th>
<th>Packages</th>
<th>Reference</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>PCA</td>
<td>Scran</td>
<td><a href="https://doi.org/10.12688/f1000research.9501.2">Lun ATL, McCarthy DJ, Marioni JC (2016). “A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor.” F1000Res., 5, 2122.</a></td>
</tr>
<tr class="even">
<td>PCA</td>
<td>Seurat</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
<tr class="odd">
<td>ICA</td>
<td>Seurat</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="tsneumap" class="section level3">
<h3 class="hasAnchor">
<a href="#tsneumap" class="anchor"></a>tSNE/UMAP</h3>
<div id="implemented-algorithms-from-packages-1" class="section level4">
<h4 class="hasAnchor">
<a href="#implemented-algorithms-from-packages-1" class="anchor"></a>Implemented Algorithms from Packages:</h4>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>Packages</th>
<th>Reference</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>tSNE</td>
<td>?</td>
<td>?</td>
</tr>
<tr class="even">
<td>UMAP</td>
<td>?</td>
<td>?</td>
</tr>
<tr class="odd">
<td>tSNE</td>
<td>Seurat</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
<tr class="even">
<td>UMAP</td>
<td>Seurat</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="visualizations-supported" class="section level3">
<h3 class="hasAnchor">
<a href="#visualizations-supported" class="anchor"></a>Visualizations Supported</h3>
<table class="table">
<thead><tr class="header">
<th>Method</th>
<th>2-Dimensional Component Plot</th>
<th>Elbow Plot</th>
<th>JackStraw Plot</th>
<th>Heatmap Plot</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>PCA</td>
<td><span class="math inline">\(\checkmark\)</span></td>
<td><span class="math inline">\(\checkmark\)</span></td>
<td><span class="math inline">\(\checkmark\)</span></td>
<td><span class="math inline">\(\checkmark\)</span></td>
</tr>
<tr class="even">
<td>ICA</td>
<td><span class="math inline">\(\checkmark\)</span></td>
<td>x</td>
<td>x</td>
<td><span class="math inline">\(\checkmark\)</span></td>
</tr>
<tr class="odd">
<td>tSNE</td>
<td><span class="math inline">\(\checkmark\)</span></td>
<td>x</td>
<td>x</td>
<td>x</td>
</tr>
<tr class="even">
<td>UMAP</td>
<td><span class="math inline">\(\checkmark\)</span></td>
<td>x</td>
<td>x</td>
<td>x</td>
</tr>
</tbody>
</table>
</div>
<div id="general-worflow" class="section level3">
<h3 class="hasAnchor">
<a href="#general-worflow" class="anchor"></a>General Worflow</h3>
<p><img src="ui_screenshots/dimensionality_reduction/dr_workflow.png"></p>
<hr>
</div>
</div>
<div id="workflow-guide" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow-guide" class="anchor"></a>Workflow Guide</h2>
<p>In general, both sub-tabs offer options for selection of data items and choice of parameters on the left side, and a visualization panel on the right side of the interface. A detailed workflow guide to run and visualize dimensionality reduction (DR) algorithms is described below:</p>
<p><img src="ui_screenshots/dimensionality_reduction/dr_1.png"> 1. To begin the DR workflow, click on the “Feature Selection &amp; Dimensionality Reduction” tab from the top menu. This workflow assumes that before proceeding towards computation of DR, data has been uploaded, filtered and normalized (and optionally variable features have been identified) through the preceding tabs. <img src="ui_screenshots/dimensionality_reduction/dr_2.png"> 2. Select “Dimensionality Reduction” tab. <img src="ui_screenshots/dimensionality_reduction/dr_3.png"> 3. Select “PCA/ICA” or “tSNE/UMAP” appropriately. <img src="ui_screenshots/dimensionality_reduction/dr_4.png"> 4. Select a data item (assay or a feature subset) which should be used for computation. <br> 5. Select an appropriate method for dimensionality reduction. Available choices are “PCA” from  package and “PCA” &amp; “ICA” from  package. <br> 6. Specify a name for the new data (reducedDim). <br> 7. Specify the number of dimensions to compute against the selected algorithm. Default value is . <br> 8. Check the boxes against the visualizations that should be plotted after computation of reducedDims. This visualizations become available after computation on the right panel. <br> 9. If “Compute HeatmapPlot?” is selected in step 8, you can specify how many features should be plotted in the heatmap by default. This setting can be changed later as well from the visualization panel on the right. <br> 10. Press “Run” to start computation. <br> 11. Once processing is complete, selected visualizations appear in this panel. <br><img src="ui_screenshots/dimensionality_reduction/dr_5.png"> 12. A 2D plot between the top two components is computed for all methods. <img src="ui_screenshots/dimensionality_reduction/dr_6.png"> 13. Elbow plot (optional) can be computed against PCA methods. It shows a relationship between the increasing number of components and the standard deviation, where components before an elbow break should be selected for downstream analysis. <br><img src="ui_screenshots/dimensionality_reduction/dr_7.png"></p>
<ol start="14" style="list-style-type: decimal">
<li>Heatmap plot panel can be used to visualize the features against each of the computed component. <br>
</li>
<li>Customizations for the heatmap plot can be made by selecting the components that should be selected. Number of columns for visualization can be specified as well for better viewing experience. <img src="ui_screenshots/dimensionality_reduction/dr_8.png">
</li>
<li>Jackstraw plot can be computed with PCA methods. <img src="ui_screenshots/dimensionality_reduction/dr_9.png">
</li>
<li>To compute tSNE or UMAP, select the “tSNE/UMAP” sub-tab from the interface. The steps below are only concerned with the computation of tSNE or UMAP and may vary between both of these methods. <br>
</li>
<li>Select a data item (assay or feature subset) to use for tSNE/UMAP computation. <br>
</li>
<li>Select method for tSNE or UMAP computation. <br>
</li>
<li>Specify the name of the new reducedDim. <br>
</li>
<li>Specify the number of components to compute with the selected method. Default value is set to . <br>
</li>
<li>Specify number of iterations. Default value is set to . <br>
</li>
<li>Set perplexity parameter for tSNE. Default value is set to code{5}. <br>
</li>
<li>Start processing of the selected method. <br>
</li>
<li>Once computation is complete, a 2D plot between the components can be visualized in this panel.</li>
</ol>
<p><em>Note: Some parameters may differ between different methods and may not have shown here.</em></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
