<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GSVA • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GSVA">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
</li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ui_gsva_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>GSVA</h1>
                        <h4 class="author">Irzam Sarfraz</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/ui_gsva.Rmd"><code>vignettes/ui_gsva.Rmd</code></a></small>
      <div class="hidden name"><code>ui_gsva.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The <strong><em>GSVA</em></strong> (Gene Set Variation Analysis) (<a href="https://doi.org/10.1186/1471-2105-14-7">Hänzelmann, S., Castelo, R. &amp; Guinney, J. GSVA: gene set variation analysis for microarray and RNA-Seq data. BMC Bioinformatics 14, 7 (2013).</a>) is a popular method for <strong><em>GSE</em></strong> (Gene Set Enrichment) and allows the identification of changes in pathway activity in RNA-Seq data. The <strong><em>Cell Annotation &amp; Pathway Analysis</em></strong> tab in the <strong><em>singleCellTK</em></strong> offers the computation and visualization of pre-selected gene sets through <strong><em>GSVA</em></strong> sub-tab. The gene sets can be selected from several databases integrated within the toolkit or can be uploaded/created by the users.</p>
</div>
<div id="general-worflow" class="section level2">
<h2 class="hasAnchor">
<a href="#general-worflow" class="anchor"></a>General Worflow</h2>
<p>The figure below describes the workflow of using <strong><em>GSVA</em></strong> through <strong><em>singleCellTK</em></strong> user interface: <img src="ui_screenshots/pathway_analysis/capa_workflow.png"></p>
</div>
<div id="workflow-guide" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow-guide" class="anchor"></a>Workflow Guide</h2>
<p>Before running the <strong><em>GSVA</em></strong> algorithm, the gene sets must be selected or uploaded through the <strong><em>Import Gene Sets</em></strong> sub-tab from the <strong><em>Data</em></strong> tab in the toolkit menu. An overall workflow guide on how to import/upload <strong><em>gene sets</em></strong> and consequently using them with <strong><em>GSVA</em></strong> sub-tab is described below. <strong><em>This workflow guide assumes that the </em>SingleCell Data* has already been uploaded through the <em>Import Single Cell Data</em></strong> sub-tab.**<br></p>
<p><img src="ui_screenshots/pathway_analysis/capa_1.png"></p>
<ol style="list-style-type: decimal">
<li><p>Select the <strong><em>Data</em></strong> tab from the <strong><em>singleCellTK</em></strong> top menu.<br></p></li>
<li><p>Select the <strong><em>Import Gene Sets</em></strong> from the drop-down menu.<br></p></li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_2.png"></p>
<ol start="3" style="list-style-type: decimal">
<li><p>Users can choose the <strong><em>Upload a GMT file</em></strong> option to upload gene sets directly stored in a GMT file or choose <strong><em>Select from a database</em></strong> option to select gene sets from multiple databases integrated within the toolkit or lastly choose the <strong><em>Paste in your gene set</em></strong> option to paste gene identifiers directly into the user-interface to create a gene set.<br></p></li>
<li><ol style="list-style-type: lower-alpha">
<li>If <strong><em>Upload a GMT file</em></strong> option is selected, users can upload a .GMT file containing the gene sets. <br>
</li>
</ol></li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_3.png"></p>
<ol start="4" style="list-style-type: decimal">
<li><ol start="2" style="list-style-type: lower-alpha">
<li>If <strong><em>Select from a database</em></strong> option is selected, the available gene sets are populated. Users can select the gene sets and click on the <strong><em>Upload</em></strong> button to get them to use in the downstream analysis. <br>
</li>
</ol></li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_4.png"></p>
<ol start="5" style="list-style-type: decimal">
<li>Once <strong><em>gene sets</em></strong> are uploaded, the selected gene sets are listed in this table. <br>
</li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_5.png"></p>
<ol start="6" style="list-style-type: decimal">
<li>To run the <strong><em>GSVA</em></strong> algorithm, go to <strong><em>Cell Annotation &amp; Pathway Analysis</em></strong> tab. <br>
</li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_6.png"></p>
<ol start="7" style="list-style-type: decimal">
<li>Select <strong><em>GSVA</em></strong> option from the drop-down menu. <br>
</li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_7.png"></p>
<ol start="8" style="list-style-type: decimal">
<li>Select <strong><em>assay</em></strong> to use with <strong><em>GSVA</em></strong>. <br>
</li>
<li>Select <strong><em>GSVA</em></strong> as the method.</li>
<li>Select <strong><em>gene sets</em></strong> to use with <strong><em>GSVA</em></strong>. Only uploaded <strong><em>gene sets</em></strong> from the <strong><em>Import Gene Sets</em></strong> page are available here. <br>
</li>
<li>Select a <strong><em>phenotype variable</em></strong> to use with the <strong><em>plots</em></strong>. <br>
</li>
<li>Select if a <strong><em>Heatmap</em></strong> should be plotted or a *<strong>Violin</strong> plot. <br>
</li>
<li>Press <strong><em>Run</em></strong> to start computation. <br>
</li>
<li>Once computation is complete, users use the <strong><em>Save Pathways</em></strong> button to store the pathways in the internal object for use in downstream analysis or use the <strong><em>Download Pathway Results</em></strong> option to download the results to the local drive. <br>
</li>
</ol>
<p><img src="ui_screenshots/pathway_analysis/capa_8.png"></p>
<ol start="15" style="list-style-type: decimal">
<li>Results can be visualized by selecting the <strong><em>Plot</em></strong> tab <strong>(16)</strong> or the <strong><em>Results Table</em></strong> tab <strong>(17)</strong>. <br>
</li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
