<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Normalization • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Normalization">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
</li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
</li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ui_normalization_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Normalization</h1>
                        <h4 class="author">Irzam Sarfraz</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/ui_normalization.Rmd"><code>vignettes/ui_normalization.Rmd</code></a></small>
      <div class="hidden name"><code>ui_normalization.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Data once uploaded and filtered through the preceding tabs can be normalized and corrected for batch-effect. This guide particularly focuses on normalization of data for downstream analysis which can be achieved through various methods and options integrated together in a single interface. Generally, users can use one of the provided normalization methods integrated from other packages, or use transformation options to manually normalize/scale data. Additionally, these transformation options can be applied to normalized data as well depending upon the type of the transformation.</p>
<div id="available-normalization-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#available-normalization-methods" class="anchor"></a>Available Normalization Methods</h3>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>Reference</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Seurat - LogNormalize</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
<tr class="even">
<td>Seurat - CLR</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
<tr class="odd">
<td>Seurat - RC</td>
<td>
<a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
<tr class="even">
<td>Seurat - SCTransform</td>
<td>
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1874-1">Normalization and variance stabilization of single-cell RNA-seq data using regularized negative binomial regression</a>, <a href="https://www.nature.com/articles/nbt.4096">Butler et al., Nature Biotechnology 2018</a> &amp; <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8">Stuart<em>, Butler</em>, et al., Cell 2019</a>
</td>
</tr>
<tr class="odd">
<td>Scater - LogNormCounts</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btw777">McCarthy DJ, Campbell KR, Lun ATL, Willis QF (2017). “Scater: pre-processing, quality control, normalisation and visualisation of single-cell RNA-seq data in R.” Bioinformatics, 33, 1179-1186.</a></td>
</tr>
<tr class="even">
<td>Scater - CPM</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btw777">McCarthy DJ, Campbell KR, Lun ATL, Willis QF (2017). “Scater: pre-processing, quality control, normalisation and visualisation of single-cell RNA-seq data in R.” Bioinformatics, 33, 1179-1186.</a></td>
</tr>
</tbody>
</table>
</div>
<div id="available-transformation-options" class="section level3">
<h3 class="hasAnchor">
<a href="#available-transformation-options" class="anchor"></a>Available Transformation Options</h3>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Option</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Log Transform</td>
<td>Simple log base 2 transformation</td>
</tr>
<tr class="even">
<td>log1p</td>
<td>Natural log transformation and additionally adds 1 to remove zeros</td>
</tr>
<tr class="odd">
<td>Z-Score</td>
<td>Standard z-score scaling</td>
</tr>
<tr class="even">
<td>trim</td>
<td>Trim values based on an upper and lower limits (can be applied with all of the above methods)</td>
</tr>
</tbody>
</table>
</div>
<div id="general-workflow" class="section level3">
<h3 class="hasAnchor">
<a href="#general-workflow" class="anchor"></a>General Workflow</h3>
<p><img src="ui_screenshots/normalization/normalization_workflow.png"></p>
</div>
</div>
<div id="workflow-guide" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow-guide" class="anchor"></a>Workflow Guide</h2>
<p>Normalization tab can be opened up by clicking on the <em>Normalization &amp; Batch Correction</em> from the top menu and further selecting the <em>Normalization</em> sub-tab in the subsequent window as shown below.</p>
<p><img src="ui_screenshots/normalization/normalization_1_select_menu.png"></p>
<p><img src="ui_screenshots/normalization/normalization_2_overview_1.png"></p>
<p>The <em>Normalization</em> user-interface is divided into three parts, a) <em>Normalization Options</em>, b) <em>Assay Options</em>, and c) <em>Available Assays</em>.</p>
<p><img src="ui_screenshots/normalization/normalization_2_overview_2.png"></p>
<div id="a-normalization-options" class="section level4">
<h4 class="hasAnchor">
<a href="#a-normalization-options" class="anchor"></a>a) Normalization Options</h4>
<p>In the <em>Normalization Options</em>, users can select a method for normalization from <code>Scater - CPM</code> or one of the methods from <code>Seurat</code> i.e. <code>LogNormalize</code>, <code>CLR</code> (Centered Log Ratio), <code>RC</code> (Relative Counts) and <code>SCTransform</code>. Users can also set a manual <em>Scaling Factor</em> (numeric value multiplied by the data values) and specify the <em>Assay Name</em> for the new <code>assay</code>.</p>
<p><img src="ui_screenshots/normalization/normalization_3_seurat.png"></p>
<p><img src="ui_screenshots/normalization/normalization_4_cpm.png"></p>
</div>
<div id="b-assay-options" class="section level4">
<h4 class="hasAnchor">
<a href="#b-assay-options" class="anchor"></a>b) Assay Options</h4>
<p>In addition to the normalization methods, manually implemented transformation methods can be used to perform different transformations on the data. These transformations include <code>Log Transform</code>, <code>log1p</code> and standard <code>Z-score</code>.</p>
<p><img src="ui_screenshots/normalization/normalization_5_assay_options.png"></p>
<p>If trimming of an <code>assay</code> is required, ‘Trim Assay’ option can be enabled and lower/upper bound values can be set. The default value for upper and lower trim values is <code>10</code> and <code>-10</code> respectively.</p>
<p><img src="ui_screenshots/normalization/normalization_6_trim_assay.png"></p>
</div>
<div id="c-available-assays" class="section level4">
<h4 class="hasAnchor">
<a href="#c-available-assays" class="anchor"></a>c) Available Assays</h4>
<p>The currently available <code>assays</code> are listed in the ‘Available Assays’ window. Whenever a new <code>assay</code> is generated, the list is updated automatically.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
