<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Heatmap visualization of DEG result — plotDEGHeatmap • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Heatmap visualization of DEG result — plotDEGHeatmap" />
<meta property="og:description" content="A differential expression analysis function has to be run in advance so that
information is stored in the metadata of the input SCE object. This function
wraps plotSCEHeatmap.
A feature annotation basing on the log2FC level called &quot;regulation&quot;
will be automatically added. A cell annotation basing on the condition
selection while running the analysis called &quot;condition&quot;, and the
annotations used from colData(inSCE) while setting the condition and
covariates will also be added." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Heatmap visualization of DEG result</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/plotDEAnalysis.R'><code>R/plotDEAnalysis.R</code></a></small>
    <div class="hidden name"><code>plotDEGHeatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A differential expression analysis function has to be run in advance so that
information is stored in the metadata of the input SCE object. This function
wraps plotSCEHeatmap.
A feature annotation basing on the log2FC level called <code>"regulation"</code>
will be automatically added. A cell annotation basing on the condition
selection while running the analysis called <code>"condition"</code>, and the
annotations used from <code>colData(inSCE)</code> while setting the condition and
covariates will also be added.</p>
    </div>

    <pre class="usage"><span class='fu'>plotDEGHeatmap</span><span class='op'>(</span>
  <span class='va'>inSCE</span>,
  <span class='va'>useResult</span>,
  doLog <span class='op'>=</span> <span class='cn'>FALSE</span>,
  onlyPos <span class='op'>=</span> <span class='cn'>FALSE</span>,
  log2fcThreshold <span class='op'>=</span> <span class='fl'>0.25</span>,
  fdrThreshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  useAssay <span class='op'>=</span> <span class='cn'>NULL</span>,
  featureAnnotations <span class='op'>=</span> <span class='cn'>NULL</span>,
  cellAnnotations <span class='op'>=</span> <span class='cn'>NULL</span>,
  featureAnnotationColor <span class='op'>=</span> <span class='cn'>NULL</span>,
  cellAnnotationColor <span class='op'>=</span> <span class='cn'>NULL</span>,
  rowDataName <span class='op'>=</span> <span class='cn'>NULL</span>,
  colDataName <span class='op'>=</span> <span class='cn'>NULL</span>,
  colSplitBy <span class='op'>=</span> <span class='st'>"condition"</span>,
  rowSplitBy <span class='op'>=</span> <span class='st'>"regulation"</span>,
  title <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"MAST Result: "</span>, <span class='va'>useResult</span><span class='op'>)</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>SingleCellExperiment inherited object.
<code><a href='runMAST.html'>runMAST()</a></code> has to be run in advance.</p></td>
    </tr>
    <tr>
      <th>useResult</th>
      <td><p>character. A string specifying the <code>analysisName</code>
used when running a differential expression analysis function.</p></td>
    </tr>
    <tr>
      <th>doLog</th>
      <td><p>Logical scalar. Whether to do <code><a href='https://rdrr.io/r/base/Log.html'>log(assay + 1)</a></code>
transformation on the assay used for the analysis. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>onlyPos</th>
      <td><p>logical. Whether to only plot DEG with positive log2_FC
value. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>log2fcThreshold</th>
      <td><p>numeric. Only plot DEGs with the absolute values of
log2FC larger than this value. Default <code>1</code>.</p></td>
    </tr>
    <tr>
      <th>fdrThreshold</th>
      <td><p>numeric. Only plot DEGs with FDR value smaller than this
value. Default <code>0.05</code>.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>character. A string specifying an assay of expression value
to plot. By default the assay used for <code><a href='runMAST.html'>runMAST()</a></code> will be used.
Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>featureAnnotations</th>
      <td><p><code>data.frame</code>, with <code>rownames</code> containing
all the features going to be plotted. Character columns should be factors.
Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>cellAnnotations</th>
      <td><p><code>data.frame</code>, with <code>rownames</code> containing
all the cells going to be plotted. Character columns should be factors.
Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>featureAnnotationColor</th>
      <td><p>A named list. Customized color settings for
feature labeling. Should match the entries in the <code>featureAnnotations</code>
or <code>rowDataName</code>. For each entry, there should be a list/vector of
colors named with categories. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>cellAnnotationColor</th>
      <td><p>A named list. Customized color settings for
cell labeling. Should match the entries in the <code>cellAnnotations</code> or
<code>colDataName</code>. For each entry, there should be a list/vector of colors
named with categories. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>rowDataName</th>
      <td><p>character. The column name(s) in <code>rowData</code> that need
to be added to the annotation. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>colDataName</th>
      <td><p>character. The column name(s) in <code>colData</code> that need
to be added to the annotation. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>colSplitBy</th>
      <td><p>character. Do semi-heatmap based on the grouping of
this(these) annotation(s). Should exist in either <code>colDataName</code> or
<code><a href='https://rdrr.io/r/base/names.html'>names(cellAnnotations)</a></code>. Default <code>"condition"</code>.</p></td>
    </tr>
    <tr>
      <th>rowSplitBy</th>
      <td><p>character. Do semi-heatmap based on the grouping of
this(these) annotation(s). Should exist in either <code>rowDataName</code> or
<code><a href='https://rdrr.io/r/base/names.html'>names(featureAnnotations)</a></code>. Default <code>"regulation"</code>.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p>character. Main title of the heatmap. Default
<code>"MAST Result: &lt;useResult&gt;"</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Other arguments passed to <code><a href='plotSCEHeatmap.html'>plotSCEHeatmap</a></code></p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code><a href='https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html'>ComplexHeatmap::Heatmap</a></code> object</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Yichen Wang</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>scExample</span>, package <span class='op'>=</span> <span class='st'>"singleCellTK"</span><span class='op'>)</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='subsetSCECols.html'>subsetSCECols</a></span><span class='op'>(</span><span class='va'>sce</span>, colData <span class='op'>=</span> <span class='st'>"type != 'EmptyDroplet'"</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='runDEAnalysis.html'>runDEAnalysis</a></span><span class='op'>(</span>inSCE <span class='op'>=</span> <span class='va'>sce</span>, method <span class='op'>=</span> <span class='st'>"DESeq2"</span>,
                     groupName1 <span class='op'>=</span> <span class='st'>"Sample1"</span>, groupName2 <span class='op'>=</span> <span class='st'>"Sample2"</span>,
                     index1 <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>100</span>, index2 <span class='op'>=</span> <span class='fl'>101</span><span class='op'>:</span><span class='fl'>190</span>,
                     analysisName <span class='op'>=</span> <span class='st'>"DESeq2"</span><span class='op'>)</span>
<span class='fu'>plotDEGHeatmap</span><span class='op'>(</span><span class='va'>sce</span>, useResult <span class='op'>=</span> <span class='st'>"DESeq2"</span>, fdrThreshold <span class='op'>=</span> <span class='fl'>1</span>, doLog <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='op'>}</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


