<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Apply ComBat-Seq batch effect correction method to SingleCellExperiment
object — runComBatSeq • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Apply ComBat-Seq batch effect correction method to SingleCellExperiment
object — runComBatSeq" />
<meta property="og:description" content="The ComBat-Seq batch adjustment approach assumes that batch effects represent
non-biological but systematic shifts in the mean or variability of genomic
features for all samples within a processing batch. It uses either parametric
or non-parametric empirical Bayes frameworks for adjusting data for batch
effects." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Apply ComBat-Seq batch effect correction method to SingleCellExperiment
object</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/runBatchCorrection.R'><code>R/runBatchCorrection.R</code></a></small>
    <div class="hidden name"><code>runComBatSeq.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The ComBat-Seq batch adjustment approach assumes that batch effects represent
non-biological but systematic shifts in the mean or variability of genomic
features for all samples within a processing batch. It uses either parametric
or non-parametric empirical Bayes frameworks for adjusting data for batch
effects.</p>
    </div>

    <pre class="usage"><span class='fu'>runComBatSeq</span><span class='op'>(</span>
  <span class='va'>inSCE</span>,
  useAssay <span class='op'>=</span> <span class='st'>"counts"</span>,
  batch <span class='op'>=</span> <span class='st'>"batch"</span>,
  covariates <span class='op'>=</span> <span class='cn'>NULL</span>,
  bioCond <span class='op'>=</span> <span class='cn'>NULL</span>,
  useSVA <span class='op'>=</span> <span class='cn'>FALSE</span>,
  assayName <span class='op'>=</span> <span class='st'>"ComBatSeq"</span>,
  shrink <span class='op'>=</span> <span class='cn'>FALSE</span>,
  shrinkDisp <span class='op'>=</span> <span class='cn'>FALSE</span>,
  nGene <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>SingleCellExperiment inherited object. Required.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>A single character indicating the name of the assay requiring
batch correction. Default <code>"counts"</code>.</p></td>
    </tr>
    <tr>
      <th>batch</th>
      <td><p>A single character indicating a field in
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></code> that annotates the batches.
Default <code>"batch"</code>.</p></td>
    </tr>
    <tr>
      <th>covariates</th>
      <td><p>A character vector indicating the fields in
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></code> that annotates other covariates,
such as the cell types. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>bioCond</th>
      <td><p>A single character indicating a field in
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></code> that annotates the biological
conditions. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>useSVA</th>
      <td><p>A logical scalar. Whether to estimate surrogate variables and
use them as an empirical control. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>assayName</th>
      <td><p>A single characeter. The name for the corrected assay. Will
be saved to <code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></code>. Default
<code>"ComBat"</code>.</p></td>
    </tr>
    <tr>
      <th>shrink</th>
      <td><p>A logical scalar. Whether to apply shrinkage on parameter
estimation. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>shrinkDisp</th>
      <td><p>A logical scalar. Whether to apply shrinkage on dispersion.
Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>nGene</th>
      <td><p>An integer. Number of random genes to use in empirical Bayes
estimation, only useful when <code>shrink</code> is set to <code>TRUE</code>. Default
<code>NULL</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The input SingleCellExperiment object with
<code>assay(inSCE, assayName)</code> updated.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>For the parameters <code>covariates</code> and <code>useSVA</code>, when the cell type
information is known, it is recommended to specify the cell type annotation
to the argument <code>covariates</code>; if the cell types are unknown but
expected to be balanced, it is recommended to run with default settings, yet
informative covariates could still be useful. If the cell types are unknown
and are expected to be unbalanced, it is recommended to set <code>useSVA</code>
to <code>TRUE</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'sceBatches'</span>, package <span class='op'>=</span> <span class='st'>'singleCellTK'</span><span class='op'>)</span>
<span class='co'># Cell type known</span>
<span class='va'>sceBatches</span> <span class='op'>&lt;-</span> <span class='fu'>runComBatSeq</span><span class='op'>(</span><span class='va'>sceBatches</span>, <span class='st'>"counts"</span>, <span class='st'>"batch"</span>,
                           covariates <span class='op'>=</span> <span class='st'>"cell_type"</span>,
                           assayName <span class='op'>=</span> <span class='st'>"ComBat_cell_seq"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Found 2 batches
#&gt; Using null model in ComBat-seq.
#&gt; Adjusting for 1 covariate(s) or covariate level(s)
#&gt; Estimating dispersions
#&gt; Fitting the GLM model
#&gt; Shrinkage off - using GLM estimates for parameters
#&gt; Adjusting the data</div><div class='input'><span class='co'># Cell type unknown but balanced</span>
<span class='va'>sceBatches</span> <span class='op'>&lt;-</span> <span class='fu'>runComBatSeq</span><span class='op'>(</span><span class='va'>sceBatches</span>, <span class='st'>"counts"</span>, <span class='st'>"batch"</span>,
                           assayName <span class='op'>=</span> <span class='st'>"ComBat_seq"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Found 2 batches
#&gt; Using null model in ComBat-seq.
#&gt; Adjusting for 0 covariate(s) or covariate level(s)
#&gt; Estimating dispersions
#&gt; Fitting the GLM model
#&gt; Shrinkage off - using GLM estimates for parameters
#&gt; Adjusting the data</div><div class='input'><span class='co'># Cell type unknown and unbalanced</span>
<span class='va'>sceBatches</span> <span class='op'>&lt;-</span> <span class='fu'>runComBatSeq</span><span class='op'>(</span><span class='va'>sceBatches</span>, <span class='st'>"counts"</span>, <span class='st'>"batch"</span>,
                           useSVA <span class='op'>=</span> <span class='cn'>TRUE</span>,
                           assayName <span class='op'>=</span> <span class='st'>"ComBat_sva_seq"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Number of significant surrogate variables is:  2 
#&gt; Iteration (out of 5 ):1  2  3  4  5  Found 2 batches
#&gt; Using null model in ComBat-seq.
#&gt; Adjusting for 2 covariate(s) or covariate level(s)
#&gt; Estimating dispersions
#&gt; Fitting the GLM model
#&gt; Shrinkage off - using GLM estimates for parameters
#&gt; Adjusting the data</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


