<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Detecting contamination with DecontX. — runDecontX • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Detecting contamination with DecontX. — runDecontX" />
<meta property="og:description" content="A wrapper function for decontX. Identify
 potential contamination from experimental factors such as ambient RNA." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Detecting contamination with DecontX.</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/celda_decontX.R'><code>R/celda_decontX.R</code></a></small>
    <div class="hidden name"><code>runDecontX.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A wrapper function for <a href='https://rdrr.io/pkg/celda/man/decontX.html'>decontX</a>. Identify
 potential contamination from experimental factors such as ambient RNA.</p>
    </div>

    <pre class="usage"><span class='fu'>runDecontX</span><span class='op'>(</span>
  <span class='va'>inSCE</span>,
  sample <span class='op'>=</span> <span class='cn'>NULL</span>,
  useAssay <span class='op'>=</span> <span class='st'>"counts"</span>,
  z <span class='op'>=</span> <span class='cn'>NULL</span>,
  maxIter <span class='op'>=</span> <span class='fl'>500</span>,
  delta <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>10</span><span class='op'>)</span>,
  estimateDelta <span class='op'>=</span> <span class='cn'>TRUE</span>,
  convergence <span class='op'>=</span> <span class='fl'>0.001</span>,
  iterLogLik <span class='op'>=</span> <span class='fl'>10</span>,
  varGenes <span class='op'>=</span> <span class='fl'>5000</span>,
  dbscanEps <span class='op'>=</span> <span class='fl'>1</span>,
  seed <span class='op'>=</span> <span class='fl'>12345</span>,
  logfile <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>A <a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a> object.</p></td>
    </tr>
    <tr>
      <th>sample</th>
      <td><p>A single character specifying a name that can be found in
<code>colData(inSCE)</code> to directly use the cell annotation; or a character
vector with as many elements as cells to indicates which sample each cell
belongs to. Default NULL. <a href='https://rdrr.io/pkg/celda/man/decontX.html'>decontX</a> will be run on cells from
each sample separately.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>A string specifying which assay in the SCE to use. Default
'counts'.</p></td>
    </tr>
    <tr>
      <th>z</th>
      <td><p>Numeric or character vector. Cell cluster labels. If NULL,
PCA will be used to reduce the dimensionality of the dataset initially,
'<a href='https://rdrr.io/pkg/uwot/man/umap.html'>umap</a>' from the 'uwot' package
will be used to further reduce the dataset to 2 dimenions and
the '<a href='https://rdrr.io/pkg/dbscan/man/dbscan.html'>dbscan</a>' function from the 'dbscan' package
will be used to identify clusters of broad cell types. Default NULL.</p></td>
    </tr>
    <tr>
      <th>maxIter</th>
      <td><p>Integer. Maximum iterations of the EM algorithm. Default 500.</p></td>
    </tr>
    <tr>
      <th>delta</th>
      <td><p>Numeric Vector of length 2. Concentration parameters for
the Dirichlet prior for the contamination in each cell. The first element
is the prior for the native counts while the second element is the prior for
the contamination counts. These essentially act as pseudocounts for the
native and contamination in each cell. If <code>estimateDelta = TRUE</code>,
this is only used to produce a random sample of proportions for an initial
value of contamination in each cell. Then
<code><a href='https://rdrr.io/pkg/MCMCprecision/man/fit_dirichlet.html'>fit_dirichlet</a></code> is used to update
<code>delta</code> in each iteration.
If <code>estimateDelta = FALSE</code>, then <code>delta</code> is fixed with these
values for the entire inference procedure. Fixing <code>delta</code> and
setting a high number in the second element will force <code>decontX</code>
to be more aggressive and estimate higher levels of contamination at
the expense of potentially removing native expression.
Default <code><a href='https://rdrr.io/r/base/c.html'>c(10, 10)</a></code>.</p></td>
    </tr>
    <tr>
      <th>estimateDelta</th>
      <td><p>Boolean. Whether to update <code>delta</code> at each
iteration.</p></td>
    </tr>
    <tr>
      <th>convergence</th>
      <td><p>Numeric. The EM algorithm will be stopped if the maximum
difference in the contamination estimates between the previous and
current iterations is less than this. Default 0.001.</p></td>
    </tr>
    <tr>
      <th>iterLogLik</th>
      <td><p>Integer. Calculate log likelihood every <code>iterLogLik</code>
iteration. Default 10.</p></td>
    </tr>
    <tr>
      <th>varGenes</th>
      <td><p>Integer. The number of variable genes to use in
dimensionality reduction before clustering. Variability is calcualted using
<code><a href='https://rdrr.io/pkg/scran/man/modelGeneVar.html'>modelGeneVar</a></code> function from the 'scran' package.
Used only when z is not provided. Default 5000.</p></td>
    </tr>
    <tr>
      <th>dbscanEps</th>
      <td><p>Numeric. The clustering resolution parameter
used in '<a href='https://rdrr.io/pkg/dbscan/man/dbscan.html'>dbscan</a>' to estimate broad cell clusters.
Used only when z is not provided. Default 1.</p></td>
    </tr>
    <tr>
      <th>seed</th>
      <td><p>Integer. Passed to <a href='https://withr.r-lib.org/reference/with_seed.html'>with_seed</a>. For reproducibility,
a default value of 12345 is used. If NULL, no calls to
<a href='https://withr.r-lib.org/reference/with_seed.html'>with_seed</a> are made.</p></td>
    </tr>
    <tr>
      <th>logfile</th>
      <td><p>Character. Messages will be redirected to a file named
`logfile`. If NULL, messages will be printed to stdout.  Default NULL.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Logical. Whether to print log messages. Default TRUE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a> object with
 'decontX_Contamination' and 'decontX_Clusters' added to the
 colData slot. Additionally, the
decontaminated counts will be added as an assay called 'decontXCounts'.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>scExample</span>, package <span class='op'>=</span> <span class='st'>"singleCellTK"</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='subsetSCECols.html'>subsetSCECols</a></span><span class='op'>(</span><span class='va'>sce</span>, colData <span class='op'>=</span> <span class='st'>"type != 'EmptyDroplet'"</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'>runDecontX</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:54 2021 ... Running 'DecontX'</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Starting DecontX</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:54 2021 .. Analyzing all cells</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:54 2021 .... Generating UMAP and estimating cell types</span></div><div class='output co'>#&gt; <span class='message'>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</span></div><div class='output co'>#&gt; <span class='message'>Also defined by 'spam'</span></div><div class='output co'>#&gt; <span class='message'>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</span></div><div class='output co'>#&gt; <span class='message'>Also defined by 'spam'</span></div><div class='output co'>#&gt; <span class='message'>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</span></div><div class='output co'>#&gt; <span class='message'>Also defined by 'spam'</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 .... Estimating contamination</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 10 | converge: 0.0278</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 20 | converge: 0.01058</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 30 | converge: 0.00584</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 40 | converge: 0.004697</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 50 | converge: 0.003356</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 60 | converge: 0.002634</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 70 | converge: 0.001941</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 80 | converge: 0.001413</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 90 | converge: 0.001284</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 100 | converge: 0.00111</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 ...... Completed iteration: 104 | converge: 0.0009827</span></div><div class='output co'>#&gt; <span class='message'>Tue Apr 13 15:46:57 2021 .. Finalizing decontaminated matrix</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Completed DecontX. Total time: 3.286583 secs</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


