<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform differential expression analysis on SCE with MAST — runMAST • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Perform differential expression analysis on SCE with MAST — runMAST" />
<meta property="og:description" content="Condition specification allows two methods:
1. Index level selection. Arguments index1 and index2 will be
used.
2. Annotation level selection. Arguments class, classGroup1 and
classGroup2 will be used." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform differential expression analysis on SCE with MAST</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/runDEAnalysis.R'><code>R/runDEAnalysis.R</code></a></small>
    <div class="hidden name"><code>runMAST.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Condition specification allows two methods:
1. Index level selection. Arguments <code>index1</code> and <code>index2</code> will be
used.
2. Annotation level selection. Arguments <code>class</code>, <code>classGroup1</code> and
<code>classGroup2</code> will be used.</p>
    </div>

    <pre class="usage"><span class='fu'>runMAST</span><span class='op'>(</span>
  <span class='va'>inSCE</span>,
  useAssay <span class='op'>=</span> <span class='st'>"logcounts"</span>,
  index1 <span class='op'>=</span> <span class='cn'>NULL</span>,
  index2 <span class='op'>=</span> <span class='cn'>NULL</span>,
  class <span class='op'>=</span> <span class='cn'>NULL</span>,
  classGroup1 <span class='op'>=</span> <span class='cn'>NULL</span>,
  classGroup2 <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>analysisName</span>,
  <span class='va'>groupName1</span>,
  <span class='va'>groupName2</span>,
  covariates <span class='op'>=</span> <span class='cn'>NULL</span>,
  onlyPos <span class='op'>=</span> <span class='cn'>FALSE</span>,
  log2fcThreshold <span class='op'>=</span> <span class='cn'>NULL</span>,
  fdrThreshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  overwrite <span class='op'>=</span> <span class='cn'>FALSE</span>,
  check_sanity <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>SingleCellExperiment inherited object.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>character. A string specifying which assay to use for MAST.
The assay should be a log-transformed normalized assay. Default
<code>"logcounts"</code>.</p></td>
    </tr>
    <tr>
      <th>index1</th>
      <td><p>Any type of indices that can subset a
SingleCellExperiment inherited object by cells. Specifies
which cells are of interests. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>index2</th>
      <td><p>Any type of indices that can subset a
SingleCellExperiment inherited object by cells. specifies
the control group against those specified by <code>index1</code>. If
<code>NULL</code> when using index specification, <code>index1</code> cells will be
compared with all other cells. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>class</th>
      <td><p>A vector/factor with <code><a href='https://rdrr.io/r/base/nrow.html'>ncol(inSCE)</a></code> elements, or a character
scalar that specifies a column name of <code>colData(inSCE)</code>. Default
<code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>classGroup1</th>
      <td><p>a vector specifying which "levels" given in <code>class</code>
are of interests. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>classGroup2</th>
      <td><p>a vector specifying which "levels" given in <code>class</code>
is the control group against those specified by <code>classGroup1</code>. If
<code>NULL</code> when using annotation specification, <code>classGroup1</code> cells
will be compared with all other cells.</p></td>
    </tr>
    <tr>
      <th>analysisName</th>
      <td><p>A character scalar naming the DEG analysis. Required</p></td>
    </tr>
    <tr>
      <th>groupName1</th>
      <td><p>A character scalar naming the group of interests. Required.</p></td>
    </tr>
    <tr>
      <th>groupName2</th>
      <td><p>A character scalar naming the control group. Required.</p></td>
    </tr>
    <tr>
      <th>covariates</th>
      <td><p>A character vector of additional covariates to use when
building the model. All covariates must exist in
<code><a href='https://rdrr.io/r/base/names.html'>names(colData(inSCE))</a></code>. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>onlyPos</th>
      <td><p>Whether to only output DEG with positive log2_FC value.
Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>log2fcThreshold</th>
      <td><p>Only out put DEGs with the absolute values of log2FC
greater than this value. Default <code>0.25</code></p></td>
    </tr>
    <tr>
      <th>fdrThreshold</th>
      <td><p>Only out put DEGs with FDR value less than this
value. Default <code>0.05</code></p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>A logical scalar. Whether to overwrite result if exists.
Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>check_sanity</th>
      <td><p>Logical scalar. Whether to perform MAST's sanity check
to see if the counts are logged. Default <code>TRUE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The input SingleCellExperiment object with
<code>metadata(inSCE)$diffExp</code> updated with the results: a list named by
<code>analysisName</code>, with <code>$groupNames</code> containing the naming of the
two conditions, <code>$useAssay</code> storing the assay name that was used for
calculation, <code>$select</code> storing the cell selection indices (logical) for
each condition, <code>$result</code> storing a <code><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></code> of
the DEGs summary, and <code>$method</code> storing <code>"MAST"</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>scExample</span>, package <span class='op'>=</span> <span class='st'>"singleCellTK"</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='subsetSCECols.html'>subsetSCECols</a></span><span class='op'>(</span><span class='va'>sce</span>, colData <span class='op'>=</span> <span class='st'>"type != 'EmptyDroplet'"</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='scaterlogNormCounts.html'>scaterlogNormCounts</a></span><span class='op'>(</span><span class='va'>sce</span>, assayName <span class='op'>=</span> <span class='st'>"logcounts"</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'>runMAST</span><span class='op'>(</span>inSCE <span class='op'>=</span> <span class='va'>sce</span>, groupName1 <span class='op'>=</span> <span class='st'>"Sample1"</span>,
 groupName2 <span class='op'>=</span> <span class='st'>"Sample2"</span>, index1 <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>20</span>, index2 <span class='op'>=</span> <span class='fl'>21</span><span class='op'>:</span><span class='fl'>40</span>,
 analysisName <span class='op'>=</span> <span class='st'>"MAST"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Assuming data assay in position 1, with name et is log-transformed.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Done!</span></div><div class='output co'>#&gt; <span class='message'>Combining coefficients and standard errors</span></div><div class='output co'>#&gt; <span class='message'>Calculating log-fold changes</span></div><div class='output co'>#&gt; <span class='message'>Calculating likelihood ratio tests</span></div><div class='output co'>#&gt; <span class='message'>Refitting on reduced model...</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Done!</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


