<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Wrapper function to run any of the integrated normalization/transformation
methods in the singleCellTK. The available methods include 'LogNormalize',
'CLR', 'RC' and 'SCTransform' from Seurat, 'logNormCounts and 'CPM' from
Scater. Additionally, users can 'scale' using Z.Score, 'transform' using
log, log1p and sqrt, add 'pseudocounts' and trim the final matrices
between a range of values. — runNormalization • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Wrapper function to run any of the integrated normalization/transformation
methods in the singleCellTK. The available methods include 'LogNormalize',
'CLR', 'RC' and 'SCTransform' from Seurat, 'logNormCounts and 'CPM' from
Scater. Additionally, users can 'scale' using Z.Score, 'transform' using
log, log1p and sqrt, add 'pseudocounts' and trim the final matrices
between a range of values. — runNormalization" />
<meta property="og:description" content="Wrapper function to run any of the integrated normalization/transformation
methods in the singleCellTK. The available methods include 'LogNormalize',
'CLR', 'RC' and 'SCTransform' from Seurat, 'logNormCounts and 'CPM' from
Scater. Additionally, users can 'scale' using Z.Score, 'transform' using
log, log1p and sqrt, add 'pseudocounts' and trim the final matrices
between a range of values." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cmd_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cmd_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cnsl_import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/cnsl_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/cnsl_normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/cnsl_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/cnsl_feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/cnsl_dimensionality_reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../articles/cnsl_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/cnsl_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/cnsl_find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/cnsl_visualization.html">Visualization</a>
    </li>
    <li>
      <a href="../articles/cnsl_heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Data</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_import_data.html">Import Data</a>
        </li>
        <li>
          <a href="../articles/ui_import_genesets.html">Import Gene Sets</a>
        </li>
        <li>
          <a href="../articles/ui_import_annotation.html">Annotation</a>
        </li>
        <li>
          <a href="../articles/ui_export_data.html">Export Data</a>
        </li>
        <li>
          <a href="../articles/ui_delete_data.html">Delete Data</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">QC &amp; Filtering</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_qc.html">Quality Control</a>
        </li>
        <li>
          <a href="../articles/ui_filtering.html">Filtering</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Normalization &amp; Batch Correction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_normalization.html">Normalization</a>
        </li>
        <li>
          <a href="../articles/ui_batch_correction.html">Batch Correction</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Feature Selection &amp; Dimensionality Reduction</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_feature_selection.html">Feature Selection</a>
        </li>
        <li>
          <a href="../articles/ui_dimemsionality_reduction.html">Dimensionality Reduction</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Differential Expression &amp; Cell Type Labeling</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_differential_expression.html">Differential Expression</a>
        </li>
        <li>
          <a href="../articles/ui_find_marker.html">Find Marker</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Visualization</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/ui_cellViewer.html">cellViewer</a>
        </li>
        <li>
          <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Wrapper function to run any of the integrated normalization/transformation
methods in the singleCellTK. The available methods include 'LogNormalize',
'CLR', 'RC' and 'SCTransform' from Seurat, 'logNormCounts and 'CPM' from
Scater. Additionally, users can 'scale' using Z.Score, 'transform' using
log, log1p and sqrt, add 'pseudocounts' and trim the final matrices
between a range of values.</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/runNormalization.R'><code>R/runNormalization.R</code></a></small>
    <div class="hidden name"><code>runNormalization.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Wrapper function to run any of the integrated normalization/transformation
methods in the singleCellTK. The available methods include 'LogNormalize',
'CLR', 'RC' and 'SCTransform' from Seurat, 'logNormCounts and 'CPM' from
Scater. Additionally, users can 'scale' using Z.Score, 'transform' using
log, log1p and sqrt, add 'pseudocounts' and trim the final matrices
between a range of values.</p>
    </div>

    <pre class="usage"><span class='fu'>runNormalization</span><span class='op'>(</span>
  <span class='va'>inSCE</span>,
  useAssay <span class='op'>=</span> <span class='st'>"counts"</span>,
  outAssayName <span class='op'>=</span> <span class='st'>"customNormalizedAssay"</span>,
  normalizationMethod <span class='op'>=</span> <span class='cn'>NULL</span>,
  scale <span class='op'>=</span> <span class='cn'>FALSE</span>,
  seuratScaleFactor <span class='op'>=</span> <span class='fl'>10000</span>,
  transformation <span class='op'>=</span> <span class='cn'>NULL</span>,
  pseudocountsBeforeNorm <span class='op'>=</span> <span class='cn'>NULL</span>,
  pseudocountsBeforeTransform <span class='op'>=</span> <span class='cn'>NULL</span>,
  trim <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>Input <code>SingleCellExperiment</code> object.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>Specify the name of the assay that should be used.</p></td>
    </tr>
    <tr>
      <th>outAssayName</th>
      <td><p>Specify the name of the new output assay.</p></td>
    </tr>
    <tr>
      <th>normalizationMethod</th>
      <td><p>Specify a normalization method from `LogNormalize`,
`CLR`, `RC` and `SCTransform` from Seurat or `logNormCounts` and `CPM` from
scater packages. Default <code>NULL</code> is set which will not run any
normalization method.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>Logical value indicating if the data should be scaled using
Z.Score. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>seuratScaleFactor</th>
      <td><p>Specify the `scaleFactor` argument if a Seurat
normalization method is selected. Default is <code>10000</code>. This parameter
will not be used if methods other than seurat are selected.</p></td>
    </tr>
    <tr>
      <th>transformation</th>
      <td><p>Specify the transformation options to run on the
selected assay. Options include `log2` (base 2 log transformation),
`log1p` (natural log + 1 transformation) and `sqrt` (square root). Default
value is <code>NULL</code>, which will not run any transformation.</p></td>
    </tr>
    <tr>
      <th>pseudocountsBeforeNorm</th>
      <td><p>Specify a numeric pseudo value that should be added 
to the assay before normalization is performed. Default is <code>NULL</code>,
which will not add any value.</p></td>
    </tr>
    <tr>
      <th>pseudocountsBeforeTransform</th>
      <td><p>Specify a numeric pseudo value that should be 
added to the assay before transformation is run. Default is <code>NULL</code>,
which will not add any value.</p></td>
    </tr>
    <tr>
      <th>trim</th>
      <td><p>Specify a vector of two numeric values that should be used
as the upper and lower trim values to trim the assay between these two
values. For example, <code><a href='https://rdrr.io/r/base/c.html'>c(10,-10)</a></code> will trim the values between 10
and -10. Default is <code>NULL</code>, which will not trim the data assay.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Logical value indicating if progress messages should be
displayed to the user. Default is <code>TRUE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Output SCE object with new normalized/transformed assay stored.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>sce_chcl</span>, package <span class='op'>=</span> <span class='st'>"scds"</span><span class='op'>)</span>
<span class='va'>sce_chcl</span> <span class='op'>&lt;-</span> <span class='fu'>runNormalization</span><span class='op'>(</span>
 inSCE <span class='op'>=</span> <span class='va'>sce_chcl</span>, 
 normalizationMethod <span class='op'>=</span> <span class='st'>"SCTransform"</span>, 
 useAssay <span class='op'>=</span> <span class='st'>"counts"</span>, 
 outAssayName <span class='op'>=</span> <span class='st'>"sctCounts"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Calculating cell attributes from input UMI matrix: log_umi</span></div><div class='output co'>#&gt; <span class='message'>Variance stabilizing transformation of count matrix of size 1260 by 2000</span></div><div class='output co'>#&gt; <span class='message'>Model formula is y ~ log_umi</span></div><div class='output co'>#&gt; <span class='message'>Get Negative Binomial regression parameters per gene</span></div><div class='output co'>#&gt; <span class='message'>Using 1260 genes, 2000 cells</span></div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=======================                                               |  33%  |                                                                              |===============================================                       |  67%  |                                                                              |======================================================================| 100%</div><div class='output co'>#&gt; <span class='message'>Found 62 outliers - those will be ignored in fitting/regularization step</span></div><div class='output co'>#&gt; <span class='message'>Second step: Get residuals using fitted parameters for 1260 genes</span></div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=======================                                               |  33%  |                                                                              |===============================================                       |  67%  |                                                                              |======================================================================| 100%</div><div class='output co'>#&gt; <span class='message'>Calculating gene attributes</span></div><div class='output co'>#&gt; <span class='message'>Wall clock passed: Time difference of 13.03926 secs</span></div><div class='output co'>#&gt; <span class='message'>Determine variable features</span></div><div class='output co'>#&gt; <span class='message'>Centering data matrix</span></div><div class='output co'>#&gt; <span class='message'>Set default assay to SCTransform</span></div><div class='output co'>#&gt; <span class='message'>Normalization performed using SCTransform method.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Vidya Akavoor, Salam Alabdullatif, Shruthi Bandyadka, Emma Briars, Xinyun Cao, Sebastian Carrasco Pro, Tyler Faits, Rui Hong, Mohammed Muzamil Khan, Yusuke Koga, Anastasia Leshchyk, Irzam Sarfraz, Yichen Wang, Zhe Wang, W. Evan Johnson, Joshua David Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


